pyvo/registry/regtap.py

```diff
@@ -305,34 +494,144 @@ def access_url(self):
         """
         the URL that can be used to access the service resource.
         """
-        return self.get("access_url", decode=True)
+        # some services declare some data models using multiple
+        # identifiers; in this case, we'll get the same access URL
+        # multiple times in here.  Don't be alarmed when that happens:
+        access_urls = list(set(self["access_urls"]))
+        if len(access_urls)==1:
+            return access_urls[0]
+        else:
+            raise dalq.DALQueryError(
+                "No unique access URL.  Use get_service.")
 
     @property
     def standard_id(self):
         """
         the IVOA standard identifier
         """
-        return self.get("standard_id", decode=True)
+        standard_ids = list(set(self["standard_ids"]))
+        if len(standard_ids)==1:
+            return standard_ids[0]
+        else:
+            raise dalq.DALQueryError(
+                "This resource supports several standards ({})."
+                "  Use get_service or restrict your query using Servicetype."
+                .format(", ".join(sorted(self.access_modes()))))
+
+    def access_modes(self):
+        """
+        returns a set of interface identifiers available on
+        this resource.
+
+        For standard interfaces, get_service will return a service
+        suitable for querying if you pass in an identifier from this
+        list as the service_type.
+
+        This will ignore VOSI (infrastructure) services.
+        """
+        return set(shorten_stdid(intf.standard_id) or "web"
```

Comment:
@msdemlei - would you consider ordering the output here? https://github.com/astropy/pyvo/issues/334, I would not block merging the PR on it, but if it's something to consider ideally it should be done before it lands in a release.